javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,169 @@
+            //show today date
+            return getDate();
+        }
+        else if (commandArray[commandArray.length - 1].toLowerCase().includes('address')) {
+            //show address
+            return "hackyourfuture's address is: Enghavevej 80 2450 københavn sv";
+        }
+
+        else {
+            //call function to perform math opration
+            return getMathResult(commandArray);
+        }
+
+    }
+    else if (commandArray[0].toLowerCase() === 'add') {
+        //call function to add todo in todos list
+        return addTodo(commandArray);
+    }
+    else if (commandArray[0].toLowerCase() === 'remove') {
+        //call function to remove todo from todos list
+        return removeTodo(commandArray);
+    }
+    else if (commandArray[0].toLowerCase() === 'set') {
+        //call function to set timer
+        return setTimer(commandArray);
+    }
+    else {
+        return "Oops I don't recognize your command please follow program manual";
+    }
+    //end main program
+}
+
+/**return the todos list*/
+function getToDos(todos) {
+    let todo = '';
+    for (let i = 0; i < todos.length - 1; i++) {
+        if (todos.length === 2) {
+            todo += todos[i]
+        }
+        else {
+            todo += todos[i] + ',';
+        }
+    }
+    if (todos.length > 1) {
+        todo += ' and ' + todos[todos.length - 1];
+    }
+    else if (todos.length === 1) {
+        todo = todos[0]
+    }
+    return 'You have ' + todos.length + ' todos ' + todo;
+}
+/**perform math opration and return result */
+function getMathResult(commandArray) {
+    const operator = commandArray[3]
+    const a = parseInt(commandArray[2]);
+    const b = parseInt(commandArray[4]);
+
+    switch (operator) {
+        case '+':
+            return a + b;
+        case '-':
+            return a - b;
+        case '*':
+            return a * b;
+        case '/':
+            return a / b;
+        case '%':
+            return a % b;
+    }
+}
+//add todo to todos list
+function addTodo(commandArray) {
+    let todoToAdd = '';
+    for (let i = 1; i < commandArray.length - 3; i++) {
+        todoToAdd += commandArray[i] + ' ';
+    }
+    todos.push(todoToAdd);
+    return todoToAdd + ' added to your todo';
+}
+//remove todo from todos list
+function removeTodo(commandArray) {
+    let todoToRemove = '';
+    for (let i = 1; i < commandArray.length - 3; i++) {
+        todoToRemove += commandArray[i] + ' ';
+    }
+    for (let i = 0; i < todos.length; i++) {
+        if (todoToRemove === todos[i]) {
+            todos.splice(i, 1);
+            break;
+        }
+    }
+    return 'Removed ' + todoToRemove + ' from your todo';
+}
+
+//set timer
+function setTimer(commandArray) {
+    //set timer
+
+    const timeInMinutes = parseInt(commandArray[commandArray.length - 2])
+    const timeInSeconds = timeInMinutes * 60;
+    let myVar = setInterval(function () { myStopFunction(); }, 1000 * timeInSeconds);
```

Comment:
Also do you use Prettier for formatting? few lines have an issue with extra spaces or the code block for better readability like this
 let timer = setInterval(function () {
      myStopFunction();
    }, 1000 * timeInSeconds);.

Reply:
I did not but now I have use it.